MetaMask/metamask-mobile
# feat(2808): edit network permissions

## **Description**
add and remove permissions to network





## **Related issues**

Contributes to solve issue: https://github.com/MetaMask/MetaMask-planning/issues/2808

## **Manual testing steps**

I have added screenshot below to see how it looks.

Set these following variables in js.env, otherwise none of these screens will show.
`
export MM_NETWORK_UI_REDESIGN_ENABLED="1"
`
`
export MM_MULTICHAIN_V1_ENABLED="1"
`
`
export MM_CHAIN_PERMISSIONS="1"
`
1. Go to the in-app brower on the mobile wallet, connect to a dapp like uniswap
2. Click on the icon on top right header, a bottom sheet opens, click on the manage permissions button
3. Edit permissions, and confirm, when going back to the screen, permissions should be kept.

## **Screenshots/Recordings**


| Before, user was prompted to give permission      | After, user gave permission and is not prompted to give it    |
|--------------|--------------|
| <img width="350" alt="Screenshot 2024-04-18 at 3 56 43 PM" src="https://github.com/user-attachments/assets/5209ce57-6b39-4f37-a2f3-9a700581ff70"> |<img width="350" alt="Screenshot 2024-04-18s at 3 56 43 PM" src="https://github.com/user-attachments/assets/c40883eb-458b-418d-8183-02677d812d43"> |



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
